nexus-tweaks.user.js

```diff
@@ -32,6 +32,391 @@ function timesCharExist(x, c){
+    if (temp) {
+      person.politics = temp[1];
+    }
+
+    // character ID and name
+    temp = /href="javascript:SelectItem\('target_id','(\d+)'\)">(.*?)</.exec(charHtml)
+    if (temp) {
+      person.id = temp[1];
+      person.name = temp[2];
+    }
+
+    // character link (level, and alternate ID)
+    temp = /\(<a href="clash.php\?op=character&amp;id=(\d+)">(\d+)(.?)<\/a>\)/.exec(
+        charHtml)
+    if (temp) {
+      person.level = temp[2];
+      // UNTESTED! No example for ranks yet available
+      if (temp[3]) {
+        person.rank = temp[3];
+      }
+    }
+
+    // HP, first check if actual hit points available
+    temp = /title="(\d+)\/(\d+)\s+hit points"/.exec(
+        charHtml)
+    if (temp) {
+      // with first aid: store correct HP current and maximum values
+      person.hpVisible = true;
+      person.currentHp = parseInt(temp[1]);
+      person.maxHp = parseInt(temp[2]);
+      person.hpTooltip = person.currentHp + '/' + person.maxHp + ' HP';
+    } else {
+      // without first aid: store fake HP values for display purposes only
+      // due to separator bars, looks best at 39/29/19/9
+      person.hpVisible = false;
+      // get health estimate from health bar image
+      temp = /src="images\/g\/HealthBar_([1-4]).gif"/.exec(charHtml);
+      person.maxHp = 39;
+      switch (parseInt(temp[1])) {
+        case 1:
+          person.currentHp = 39;
+          person.hpTooltip = 'Healthy';
+          break;
+        case 2:
+          person.currentHp = 29;
+          person.hpTooltip = 'Injured';
+          break;
+        case 3:
+          person.currentHp = 19;
+          person.hpTooltip = 'Serious';
+          break;
+        case 4:
+          person.currentHp = 9;
+          person.hpTooltip = 'Critical';
+          break;
+      }
+    }
+
+    // MP
+    temp = /title="(\d+)\/(\d+)\s+magic points"/.exec(
+        charHtml)
+    if (temp) {
+      person.mpVisible = true;
+      person.currentMp = parseInt(temp[1]);
+      person.maxMp = parseInt(temp[2]);
+      person.hpTooltip += ', ' + person.currentMp + '/' + person.maxMp + ' MP';
+    } else {
+      person.currentMp = 0;
+      person.maxMp = 0;
+    }
+
+    // Morality
+    temp = /title="(\w+)\s+alignment"/.exec(charHtml)
+    if (temp) {
+      person.morality = temp[1];
+    }
+
+    // Sorcerers Might
+    temp = /SM (\d+) min/.exec(charHtml);
+    if (temp) {
+      person.smDuration = temp[1];
+      var smStatus = {
+        'abbrev': 'SM ' + person.smDuration,
+        'title': 'Sorcerer\'s Might (' + person.smDuration + ' minutes)',
+        'class': 'status-tag-negative'
+      };
+      person.statusTags.push(smStatus);
+    }
+
+    // Get all externally visible statuses (displayed as images in vanilla)
+    // This will fetch hidden, poison, agony curse etc. (see below)
+    var statusRegExp = /<img.+?title="(.+?)".+?src="(.+?)"/;
+    var statusMatch = statusRegExp.exec(charHtml);
+
+    while (statusMatch != null) {
```

Comment:
blocker: While loops should have a circuit-breaker clause, e.g. if it runs for 50 iterations it should break out.

Reply:
circuit breaker implemented with a generous 20 iterations (the actual status effects should be well under that)